src/transformers/sagemaker/training_args_sm.py

```diff
@@ -77,6 +77,11 @@ def __post_init__(self):
     @cached_property
     def _setup_devices(self) -> "torch.device":
         logger.info("PyTorch: setting up devices")
+        if torch.distributed.is_initialized() and self.local_rank == -1:
+            logger.warning(
+                "torch.distributed process group is initialized, but local_rank == -1. "
+                "In order to use Torch DDP, set local_rank or 'LOCAL_RANK' env var."
```

Comment:
```suggestion
                "In order to use Torch DDP, launch your script with `python -m torch.distributed.launch"
```
It's unlikely users will actually set those variables, the easiest way is to use the distributed launcher ;-)

Reply:
@sgugger Done, PTAL